src/lib.rs

```diff
@@ -312,7 +310,12 @@ pub mod pallet {
                 .map_err(|_| Error::InvalidAccountSize)?;
 
             let account_bytes: [u8; 32] = address.into();
-            Ok(AccountAddress::new(account_bytes))
+            let address = AccountAddress::new(account_bytes);
+            if address == CORE_CODE_ADDRESS {
+                Err(Error::<T>::AddressNotAllowedForPublishing)
```

Comment:
I am unsure if we can assume for what purpose somebody called the to_move_address conversion function.

We use this function from within balance.rs as well so it's not only for publishing.

Maybe we can call it `StdlibAddressNotAllowed`